### PR TITLE
[doc] update Thread Bourder Router documentation

### PR DESCRIPTION
**Pull request by lmaciejonczyk**
Starting from NCS v1.7 the OpenThread Bourder Router is supported
by NCS Thread radio co-processor sample and the documentation needs
to be adjusted.
Additionaly the OpenThread Bourder Router docker image for nRF Connect
SDK is provided which has been tested with corresponding version of
the NCS Thread radio co-processor.

### CODE REVIEW

File: doc/nrf/ug_thread_tools.rst [hunk 1]
```diff
@@ -79,50 +79,21 @@ As neither the Linux-based PC nor the Raspberry Pi have such radio capability, y
...
 
    .. tabs::
 
       .. tab:: nRF52840 Dongle (USB transport)
 
          .. code-block:: console
 
-            rm -rf build
-            script/build nrf52840 USB_trans -DOT_BOOTLOADER=USB -DOT_THREAD_VERSION=1.2
+            west build -b nrf52840dongle_nrf52840 -- -DOVERLAY_CONFIG="overlay-rcp.conf ../common/overlay-thread_1_2_mtd.conf overlay-usb.conf"
```
Comment (edmont) [inline doc/nrf/ug_thread_tools.rst:90]: Why the MTD overlay? Shouldn't be FTD for complete functionality? Or doesn't even matter for RCP?

Comment (lmaciejonczyk) [reply doc/nrf/ug_thread_tools.rst:90]: In general it should be the same since RCP provides only radio capabilities and the rest is depended on the host configuration. But with mtd overlay RCP image has less memory footprint.

Comment (edmont) [reply doc/nrf/ug_thread_tools.rst:90]: But there could be dependencies, like FTD required for CSL transmitter, for instance.

**Review by edmont — APPROVED** [review]
Pre-approving until the placeholders are updated.

File: doc/nrf/ug_thread_tools.rst [hunk 2]
```diff
@@ -196,6 +167,16 @@ However, this approach requires you to download the OpenThread Border Router rep
 To set up and configure the OpenThread Border Router, follow the official `OpenThread Border Router Codelab tutorial`_ on the OpenThread documentation portal.
 Omit the *Build and flash RCP firmware* section, because this section duplicates the steps performed in the previous section.
 
+.. note::
+
+   OTBR is developing every day and it may happen that the newest version is incompatible with the specific version of NCS.
+   To make sure that everything is working correctly use specific commit of OTBR which was tested with the right version of NCS.
```
Comment (wiba-nordic) [inline doc/nrf/ug_thread_tools.rst:173]:
```suggestion
   To make sure that everything is working correctly, use a specific commit of OTBR that has been tested with the corresponding version of NCS.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
Some change requests, for the most part related to the documentation building correctly.

File: doc/nrf/ug_thread_tools.rst [hunk 3]
```diff
@@ -445,61 +450,52 @@ Build the OpenThread POSIX applications by performing the following steps:
...
+* For UART transport:
 
   .. parsed-literal::
      :class: highlight
 
-     sudo ./build/posix/src/posix/ot-cli 'spinel+hdlc+uart://\ *ncp_uart_device*\ ?uart-baudrate=\ *baud_rate*' --verbose
+     'spinel+hdlc+uart://\ *ncp_uart_device*\ ?uart-baudrate=\ *baud_rate*'
 
-* For ``ot-daemon`` and ``ot-ctl``:
+   Replace the following parameters:
```
[the remark this answers is not in the document]
Comment (lmaciejonczyk) [reply doc/nrf/ug_thread_tools.rst]: @wiba-nordic could you fix indentation for these changes? Formatting is still not working for me.

**Review by wiba-nordic — APPROVED** [review]
Preview build is still broken, but both local build and downloading the cache show everything is fine.

Marking approved from my end, on the assumption that placeholders won't break anything.

File: doc/nrf/ug_thread_tools.rst [hunk 4]
```diff
@@ -196,6 +167,17 @@ However, this approach requires you to download the OpenThread Border Router rep
...
 
+.. note::
+
+   OTBR is developing every day and it may happen that the newest version is incompatible with the specific version of NCS.
+   To make sure that everything is working correctly, use a specific commit of OTBR that has been tested with the corresponding version of NCS.
+
+   +--------------------------+--------------------+
+   | OpenThread Border Router |        NCS         |
+   +==========================+====================+
+   |        e149a60eb         |       v1.7         |
```
Comment (edmont) [inline doc/nrf/ug_thread_tools.rst:178]:
```suggestion
   |        e149a60e          |       v1.7         |
```
Maybe use an even number of characters for the SHA?

Comment (lmaciejonczyk) [reply doc/nrf/ug_thread_tools.rst:178]: Why do you want to use event number?

Comment (edmont) [reply doc/nrf/ug_thread_tools.rst:178]: Never mind, just realized that GitHub itself is using 7 chars as abbreviation.

File: doc/nrf/ug_thread_tools.rst [hunk 5]
```diff
@@ -196,6 +167,17 @@ However, this approach requires you to download the OpenThread Border Router rep
...
 
+.. note::
+
+   OTBR is developing every day and it may happen that the newest version is incompatible with the specific version of NCS.
+   To make sure that everything is working correctly, use a specific commit of OTBR that has been tested with the corresponding version of NCS.
+
+   +--------------------------+--------------------+
+   | OpenThread Border Router |        NCS         |
+   +==========================+====================+
+   |         e149a60          |       v1.7         |
```
Comment (LuDuda) [inline doc/nrf/ug_thread_tools.rst:178]: I would prefer to have a simple command instead of the table. I don't think we need to maintain the table with versions for all releases anyway right? Users can switch between documentation of different releases.

So just have a sentence that: when following the CodeLab tutorial, after cloning the repository please ensure to checkout to the compatible commit id.
`git chekout e149a60`

File: doc/nrf/ug_thread_tools.rst [hunk 6]
```diff
@@ -79,50 +79,21 @@ As neither the Linux-based PC nor the Raspberry Pi have such radio capability, y
...
 
    .. tabs::
 
       .. tab:: nRF52840 Dongle (USB transport)
 
          .. code-block:: console
 
-            rm -rf build
-            script/build nrf52840 USB_trans -DOT_BOOTLOADER=USB -DOT_THREAD_VERSION=1.2
+            west build -b nrf52840dongle_nrf52840 -- -DOVERLAY_CONFIG="overlay-rcp.conf ../common/overlay-thread_1_2_ftd.conf overlay-usb.conf"
```
Comment (LuDuda) [inline doc/nrf/ug_thread_tools.rst:90]: nit: I think that `overlay-usb.conf` could be automatically included when building for `nrf52840dongle_nrf52840` but this is not related to this PR.

File: doc/nrf/ug_thread_tools.rst [hunk 7]
```diff
@@ -224,20 +206,45 @@ To install and configure the OpenThread Border Router using the Docker container
 
       sudo docker network create --ipv6 --subnet fd11:db8:1::/64 -o com.docker.network.bridge.name=otbr0 otbr
 
-#. Download the latest version of the OpenThread Border Router Docker image by running the following command:
+#. Download the latest version of the OpenThread Bourder Router docker container for nRF Connect SDK by running the following command:
```
Comment (Damian-Nordic) [inline doc/nrf/ug_thread_tools.rst:209]: 1. Typo: "Bourder"
2. It's no longer "the latest" - "compatible with this version of |NCS|"?
3. We don't download a container, we download an image based on which a container is created.

File: doc/nrf/ug_thread_tools.rst [hunk 8]
```diff
@@ -224,20 +200,45 @@ To install and configure the OpenThread Border Router using the Docker container
 
       sudo docker network create --ipv6 --subnet fd11:db8:1::/64 -o com.docker.network.bridge.name=otbr0 otbr
 
-#. Download the latest version of the OpenThread Border Router Docker image by running the following command:
+#. Download the compatible version of the OpenThread Bourder Router docker image by running the following command:
```
Comment (LuDuda) [inline doc/nrf/ug_thread_tools.rst:203]: typo in the last commit: Bourder

**Review by wiba-nordic — COMMENTED** [review]
Since this got merged, we'll need another PR later to fix the typo and to make the two improvements to the language I've noted.
